social_distribution/posts/views.py

```diff
@@ -298,6 +305,57 @@ def put(self, request, author_id, post_id):
                     return Response(serializer.errors, 400)
             else:
                 return Response({'detail': 'Post with this id already exists'}, 400)
+                
+
+
+
+@login_required
+def create_comment(request, author_id, post_id):
+    current_author = Author.objects.get(id=author_id)
+    post = get_object_or_404(Post, id=post_id)
+    comments = post.comments.all()  #get all comments from that post_id
+
+
+    if request.method == "GET":
+        return render(request, 'comments/create_comment.html', {'comments': comments, 'author':current_author, 'post':post})
+    elif request.method == "POST":
+        comment=request.POST['comment']
+        postID=request.POST['post']
+        post=Post.objects.get(id=postID) # Obtain the instance
+        author = Author.objects.get(user=request.user) # Obtain the instance
+
+        comment = Comment.objects.create(author=author, post=post, comment=comment)
+        data =[{'published': comment.published}]
+        return JsonResponse({"bool":True, 'published': comment.published})
+
+
+class CommentsAPI(APIView):
```

Comment:
you should fetch comments within posts API to reduce API call counts as specified in the spec

Reply:
I've updated my code, comments are collapsible in a post, and data is fetched within posts API now, check it out when you have time